lib/src/neu/verifications/contract-mismatcher.ts

```diff
@@ -328,6 +344,76 @@ export class ContractMismatcher {
     );
   }
 
+  private getQueryParamsArraySerializationStrategy(): { comma: boolean } {
+    const comma =
+      this.contract.config.paramSerializationStrategy.query.array === "comma";
+
+    return { comma };
+  }
+
+  private findMismatchOnRequestQueryParams(
+    endpoint: Endpoint,
+    userInputRequest: UserInputRequest
+  ): Result<Mismatch[], Error> {
+    const queryParamsString = url.parse(userInputRequest.path).query || "";
+    const contractQueryParams = endpoint.request!!.queryParams;
+
+    const queryParams = qs.parse(queryParamsString, {
+      ...this.getQueryParamsArraySerializationStrategy()
+    });
+
+    let result;
+    let mismatches: Mismatch[] = [];
+    for (const {
+      name: queryParamName,
+      optional,
+      type: contractQueryParamType
+    } of contractQueryParams) {
+      const requestQueryParam = queryParams[queryParamName];
+
+      // Query parameter is mandatory and hasn't been provided
+      if (typeof requestQueryParam === "undefined" && !optional) {
+        mismatches.push(
+          new Mismatch(
+            `Query parameter "${queryParamName}" is required but hasn't been provided.`
+          )
+        );
+      }
+
+      if (typeof requestQueryParam === "undefined") {
+        continue;
+      }
```

Comment:
nit: this is kind of hard to read as I'd need to understand the if method before this one first.
I'd prefer putting this as a nested if statement with everything below.

Not a big deal, it's just hard to read and understand the more this codebase is going to grow later on.

Reply:
hey - i prefer linear code and tend to avoid nested branching. I'll have a look and will try to make this easier to read.